py/modbuiltins.c

```diff
@@ -378,7 +378,16 @@ MP_DEFINE_CONST_FUN_OBJ_1(mp_builtin_ord_obj, mp_builtin_ord);
 STATIC mp_obj_t mp_builtin_pow(size_t n_args, const mp_obj_t *args) {
     switch (n_args) {
         case 2: return mp_binary_op(MP_BINARY_OP_POWER, args[0], args[1]);
-        default: return mp_binary_op(MP_BINARY_OP_MODULO, mp_binary_op(MP_BINARY_OP_POWER, args[0], args[1]), args[2]); // TODO optimise...
+        default:
+#if !MICROPY_INT_POW3
+            mp_raise_msg(&mp_type_NotImplementedError, "3-arg pow() not supported");
+#else
```

Comment:
Well, if I knew you'd disagree, I'd leave it alone. #elif essentially allows to have switch () - like behavior (that's also why Python doesn't have switch, its elif is enough). If you look thru the code, we also indent #if directives, i.e. we use it (in some) cases not as just preprocessor, but as meta-language (distinction is very fine).

---

Sorry, I wanted to make "multi-review", so all comments go together with final "I took care of it" note, but pressed wrong button. I'm a firm believer of contributor-maintainer cooperation, and firmly believer that first-time contributor should not be made to go thru 10 patch revisions whenever possible (a maintainer can take some of them on themselves).